packages/appium/lib/extension/manifest.js

```diff
@@ -156,51 +156,117 @@ export class Manifest {
+   *
    * @returns {Promise<boolean>} `true` if any extensions were added, `false` otherwise.
    */
   async syncWithInstalledExtensions() {
-    // this could be parallelized, but we can't use fs.walk as an async iterator
+    /**
+     * "Changed" flag. If this method does anything to the internal data, this
+     * will be set to `true`.  This value is returned from this method.
+     */
     let didChange = false;
 
+    /**
+     * Update `didChange` if the value is truthy
+     * @param {any} value
+     * @returns {void}
+     */
+    const setChangedFlag = (value) => {
+      didChange = Boolean(value) || didChange;
+    };
+
     /**
      * Listener for the `match` event of a `glob` instance
      * @param {string} filepath - Path to a `package.json`
      * @returns {Promise<void>}
      */
-    const onMatch = async (filepath) => {
+    const handlePackage = async (filepath) => {
       try {
         const pkg = JSON.parse(await fs.readFile(filepath, 'utf8'));
         if (isExtension(pkg)) {
-          const extType = isDriver(pkg) ? DRIVER_TYPE : PLUGIN_TYPE;
-          /**
-           * this should only be 'unknown' if the extension's `package.json` is invalid
-           * @type {string}
-           */
-          const name = isDriver(pkg)
-            ? pkg.appium.driverName
-            : isPlugin(pkg)
-            ? pkg.appium.pluginName
-            : '(unknown)';
-          if (
-            (isDriver(pkg) && !this.hasDriver(name)) ||
-            (isPlugin(pkg) && !this.hasPlugin(name))
-          ) {
+          /** @type {ExtensionType} */
+          let extType;
+          /** @type {string} */
+          let name;
+          /** @type {boolean} */
+          let isNew;
+
+          if (isDriver(pkg)) {
+            extType = DRIVER_TYPE;
+            name = pkg.appium.driverName;
+            isNew = !this.hasDriver(name);
+          } else if (isPlugin(pkg)) {
+            extType = PLUGIN_TYPE;
+            name = pkg.appium.pluginName;
+            isNew = !this.hasPlugin(name);
+          } else {
+            log.warn(
+              `Discovered possible extension in package "${pkg.name}" but it is invalid. Try reinstalling it or updating to a new version.`
+            );
+            return;
+          }
+
+          if (isNew) {
             log.info(`Discovered installed ${extType} "${name}"`);
           }
-          const changed = this.addExtensionFromPackage(pkg, filepath);
-          didChange = didChange || changed;
+
+          try {
+            setChangedFlag(this.addExtensionFromPackage(pkg, filepath));
+          } finally {
+            // remove the extension from the missing list, if it's there
+            if (isDriver(pkg)) {
+              missingDrivers.delete(pkg.name);
+            } else {
+              missingPlugins.delete(pkg.name);
+            }
+          }
         }
-      } catch {}
+      } catch (err) {
+        // only eat ENOENT errors in which `package.json` is missing.
```

Comment:
to me this looks like we are eating all errors, but choosing to log non-ENOENT errors. the idea is that we never want this to fail in any situation?

Reply:
right. if there's no package.json we don't really care.  if the error is something else it may be interesting

Reply:
but not interesting enough to abort operation--a warning is better than nothing. I don't consider that "eating"

Reply:
ok, i guess i did. just want to make sure the comment reflects your intentions